lib/geometry/polygon.nit

```diff
@@ -0,0 +1,428 @@
+			return true
+		else
+			return false
+		end
+	end
+end
+
+private class Projection
+	var min: Float is writable
+	var max: Float is writable
+
+	fun overlap(other: Projection): Bool do
+		return not (min > other.max or other.min > max)
+	end
+end
+
+private class PointXCompare
+	super Comparator
+
+	redef type COMPARED: Point[Float]
+
+	redef fun compare(a,b : COMPARED): Int do
+		if a.x == b.x then
+			if a.y == b.y then return 0
+			if a.y > b.y then return - 1
+			return 1
+		else
+			if a.x > b.x then return -1
+			return 1
+		end
+	end
+end
+
+# Sorter for polygon vertices
+private abstract class PolygonSorter
+	super Comparator
+
+	redef type COMPARED: Point[Float]
+
+	# Center of the polygon's points
+	var center: COMPARED
+
+	# init calculating the center
+	init with_center(pts : Array[Array[Float]]) do
+		center = calc_center(pts)
+	end
+
+	# Calculate the center
+	fun calc_center(pts : Array[Array[Float]]): COMPARED do
+		var sumx = 0.0
+		var sumy = 0.0
+		for ap in pts do
+			sumx += ap[0]
+			sumy += ap[1]
+		end
+		return new Point[Float](sumx / pts.length.to_f, sumy / pts.length.to_f)
+	end
+end
+
+# AntiClockSort
+private class AntiClockSort
+	super PolygonSorter
+
+	redef fun compare(a,b: COMPARED): Int do
+		if a.x == b.x and a.y == b.y then return 0
+		if a.x - center.x >= 0.0 and b.x - center.x < 0.0 then return -1
+		if a.x - center.x < 0.0 and b.x - center.x >= 0.0 then return 1
+		if a.x - center.x == 0.0 and b.x - center.x == 0 then
+			if a.y - center.y >= 0.0 or b.y - center.y >= 0.0 then
+				if a.y > b.y then return -1
+				return 1
+			end
+			if b.y > a.y then return -1
+			return 1
+		end
+
+		var det = (a.x - center.x) * (b.y - center.y) - (b.x - center.x) * (a.y - center.y)
+		if det > 0.0 then return 1
+		if det < 0.0 then return -1
+
+		var d1 = (a.x - center.x) * (a.x - center.x) + (a.y - center.y) * (a.y - center.y)
+		var d2 = (b.x - center.x) * (b.x - center.x) + (b.y - center.y) * (b.y - center.y)
+		if d1 > d2 then return -1
+		return 1
+	end
+end
+
+# ClockSort
+private class ClockSort
+	super PolygonSorter
+
+	redef fun compare(a,b: COMPARED): Int do
+		if a.x == b.x and a.y == b.y then return 0
+		if a.x - center.x >= 0.0 and b.x - center.x < 0.0 then return 1
+		if a.x - center.x < 0.0 and b.x - center.x >= 0.0 then return -1
+		if a.x - center.x == 0.0 and b.x - center.x == 0 then
+			if a.y - center.y >= 0.0 or b.y - center.y >= 0.0 then
+				if a.y > b.y then return 1
+				return -1
+			end
+			if b.y > a.y then return 1
+			return -1
+		end
+
+		var det = (a.x - center.x) * (b.y - center.y) - (b.x - center.x) * (a.y - center.y)
+		if det > 0.0 then return -1
+		if det < 0.0 then return 1
+
+		var d1 = (a.x - center.x) * (a.x - center.x) + (a.y - center.y) * (a.y - center.y)
+		var d2 = (b.x - center.x) * (b.x - center.x) + (b.y - center.y) * (b.y - center.y)
+		if d1 > d2 then return 1
+		return -1
+	end
+end
+
+
+# Return the convex hull of the set of points represented by `points`
```

Comment:
You can do a better use of the parameter name: Get the convex hull of the set of `points`